### PR TITLE
Fix #1903: Remove noWarnMissing option from Linker

### LINKED ISSUE #1903: Remove `noWarnMissing` option of linker
The option was originally introduced to allow to run partests which do not link properly. Since we support much more of the Java lib now, this should be less of an issue.

I'll investigate the impact on partests.

### PR DESCRIPTION
**Pull request by gzm0**
This breaks binary compatibility of the external tools API (as scheduled for 0.6.6).

### CODE REVIEW

**Comment by gzm0** [comment]
Review by @sjrd

File: tools/jvm/src/main/scala/org/scalajs/core/tools/optimizer/ScalaJSClosureOptimizer.scala [hunk 1]
```diff
@@ -290,48 +230,15 @@ object ScalaJSClosureOptimizer {
...
-
-    // For binary compatibility
-    override def equals(other: Any): Boolean = super.equals(other)
-
-    // For binary compatibility
-    override def hashCode(): Int = super.hashCode()
-
-    // For binary compatibility
-    override def toString(): String =
-      productArray.mkString("Config(", ", ", ")")
```
Comment (sjrd) [inline tools/jvm/src/main/scala/org/scalajs/core/tools/optimizer/ScalaJSClosureOptimizer.scala]: Maybe we want to actually keep a meaningful-ish `toString()`.

File: tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/ScalaJSOptimizer.scala [hunk 2]
```diff
@@ -320,48 +256,15 @@ object ScalaJSOptimizer {
...
-
-    // For binary compatibility
-    override def equals(other: Any): Boolean = super.equals(other)
-
-    // For binary compatibility
-    override def hashCode(): Int = super.hashCode()
-
-    // For binary compatibility
-    override def toString(): String =
-      productArray.mkString("Config(", ", ", ")")
```
Comment (sjrd) [inline tools/shared/src/main/scala/org/scalajs/core/tools/optimizer/ScalaJSOptimizer.scala]: Same here about `toString()`

**Comment by sjrd** [comment]
That's all, good job :-)

I'm surprised you didn't introduce a big all-caps

> I DELETE CODE

in the PR description :p

**Comment by gzm0** [comment]
Updated

**Comment by sjrd** [comment]
LGTM

**Comment by gzm0** [comment]
Update. One of the buglists was wrong (2.12.0-M2).

**Pull request merged**